### ISSUE TITLE
Test Accuracy Drop with Permuted Model

### ISSUE THREAD

**Issue by MaxH1996** [samuela/git-re-basin#3]
Hi, I am currently trying to run some of the code on my own experiments and I am encountering somewhat of a strange problem. When I use weight matching on two trained ResNet20 models with 4x width-multiplier (CIFAR-10), the test accuracy of the permuted model drops significantly, as shown in the plot below. 
![output](https://user-images.githubusercontent.com/80034375/192537963-4c242e0d-74db-407a-9175-24adb06cf001.png)
I was wondering if you have encountered this behavior before. I guess I went wrong somewhere, but I am not quite sure what exactly is causing this problem. 

Thanks!

**Comment by samuela** [comment]
This indicates a bug to me, since the permuted model, pi(B), should behave exactly the same as the original model B. Can you test if B and pi(B) have equal outputs on equivalent batches?

**Comment by MaxH1996** [comment]
I checked the outputs and they were not equal. I went back to check the ResNet20 I was using, and although it seemed like a pretty standard implementation, it had some slight deviations from the ResNet used in your code. These deviations were apparently the root cause. I changed to the ResNet20 used in the PyTorch implementation of Git Re-basin and now it behaves as expected. 

Not sure though what was wrong with the ResNet I used. I'll look into that.

**Comment by themrzmaster** [comment]
maybe you were using batchnorm?

**Comment by adampan0527** [comment]
> This indicates a bug to me, since the permuted model, pi(B), should behave exactly the same as the original model B. Can you test if B and pi(B) have equal outputs on equivalent batches?

Hi! I met same problem recently, using ResNet20 models with 16x width-multiplier, and my pi(B) have same performance with B, but when I do simple average to merge A and pi(B) is even worse than A and B, which is simple_average(A, pi(B))<simple_average(A, B), while averaging A and pi(B) should have better performance.
Is this only happens on ResNet20?
